src/main/java/org/medicmobile/webapp/mobile/SettingsStore.java

```diff
@@ -27,7 +27,11 @@ public String getUrlToLoad(Uri url) {
 	}
 
 	public boolean isRootUrl(String url) {
-		return getAppUrl().equals(url);
+		if (url == null) {
+			return false;
+		}
+
+		return getAppUrl().equals(url.trim().replaceAll("/$", ""));
```

Comment:
Is this what is actually causing the bug around the "Connection Error Page"?  Is it just a race condition between hitting the `UrlHandler.onReceivedError` before the AppUrlVerifier task could save the cleaned URL?

Also, for the sake of code reusibility, does it make sense to make the `AppUrlVerifier.clean` method `public static`? Then we could use the same cleaning logic here...